contracts/zero-ex/contracts/src/external/Puppet.sol

```diff
@@ -0,0 +1,155 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/OwnableV06.sol";
+import "../errors/LibPuppetRichErrors.sol";
+import "./IPuppet.sol";
+
+
+/// @dev A contract that can execute arbitrary calls from its owner.
+contract Puppet is
+    IPuppet,
+    OwnableV06
```

Comment:
Probably going to ditch `Ownable` and storing the owner in an `immutable` so this contract is 100% stateless.

Reply:
I did it.